src/app/content/components/Content.screenshotspec.tsx

```diff
@@ -0,0 +1,43 @@
+/** @jest-environment puppeteer */
+import { Page } from 'puppeteer';
+import { finishRender, navigate, setDesktopViewport, setMobileViewport } from '../../../test/browserutils';
+import { Config } from '../../../test/matchers/toMatchImageSnapshot';
+
+const TEST_PAGE_NAME = 'test-page-for-generic-styles';
+const TEST_ANCHOR = 'term103';
+const TEST_PAGE_URL = `/books/book-slug-1/pages/${TEST_PAGE_NAME}#${TEST_ANCHOR}`;
+const SNAPSHOT_OPTIONS: Config = {
+  CI: {
+    failureThreshold: 1.5,
+    failureThresholdType: 'percent',
+  },
+};
+
+describe('Content', () => {
+  const testCases: { [testCase: string]: (target: Page) => Promise<void> } = {
+    Desktop: setDesktopViewport, Mobile: setMobileViewport,
+  };
+
+  for (const testCase of Object.keys(testCases)) {
+    describe(testCase, () => {
+      beforeEach(() => {
+        const setViewport = testCases[testCase];
+        setViewport(page);
+      });
+
+      it('scrolls correctly to all elements', async() => {
+        await navigate(page, TEST_PAGE_URL);
+        await finishRender(page);
+        expect(await page.screenshot()).toMatchImageSnapshot(SNAPSHOT_OPTIONS);
```

Comment:
can you do these as browserspecs by measuring the page scrollTop after every scroll against a known or computed-good value? the screenshots are brittle and add a lot of bytes to the repo. i prefer to use the screenshots when something just can't be reasonably tested with math (like: `it('looks good')`)

Reply:
Alright, I'll change them to browserspecs.